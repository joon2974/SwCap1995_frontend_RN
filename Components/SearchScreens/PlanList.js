/* eslint-disable react/no-access-state-in-setstate */
import React, { Component } from 'react';
import {
  View,
  StyleSheet,
  Dimensions,
  ImageBackground,
  TouchableOpacity,
  FlatList,
} from 'react-native';
import { CardFive } from './Cards';

const { width } = Dimensions.get('window');

export default class PlanList extends Component {
    state = {
      data: [],
      nowPage: 1,
      moreData: 0,
      detailedCategory: null,
      // nowRecommended: this.props.route.params,  이거 마지막으로 api연동 해줘야함
    }

    async componentDidMount() {
      // this.getData();
      await this.setState({ detailedCategory: this.props.route.params.detailedCategory });
      this.getData();
    }

    
  getData = () => {
    if (this.state.moreData === 0) {
      const url = 'http://49.50.172.58:3000/plans/search?query=' + this.state.detailedCategory + '&limit=10&page=' + this.state.nowPage;
      fetch(url)
        .then((r) => r.json())
        .then((data) => {
          this.setState({ 
            data: this.state.data.concat(data.plans),
            nowPage: this.state.nowPage + 1,
          });
          if (data.plans.length === 0) {
            this.setState({ moreData: 1 });
          }
        });
    }
  }

  
    handleLoadMore = () => {
      this.getData();
    }
  
    renderItem = ({ item }) => (
      <TouchableOpacity style={{}} onPress={() => this.props.navigation.navigate('플랜 상세 정보', { item: item })}>
        <CardFive
          title={item.title}
          subTitle={item.category}
          image={{
            uri: item.image_url,
          }}
        />
      </TouchableOpacity>
    );
    

    render() {
      return (
        <View style={styles.container}>
          <ImageBackground source={require('../../imgs/backReverse8.png')} style={{ width: width }}>

            <FlatList 
              style={{ marginTop: 10, width: width }}
              data={this.state.data}
              renderItem={this.renderItem}
              keyExtractor={(item) => item.id.toString()}
              onEndReached={this.handleLoadMore}
              onEndReachedThreshold={1}
              ref={(ref) => { this.flatList = ref; }}
          />

          </ImageBackground>
        </View>
      );
    }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: width * 1,
    backgroundColor: 'white',
    justifyContent: 'center',
    alignItems: 'center',
    
  }, 
});
