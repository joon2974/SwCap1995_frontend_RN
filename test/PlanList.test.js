import React from 'react';
import renderer from 'react-test-renderer';

import PlanList from '../Components/SearchScreens/PlanList';

describe('<PlanList />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<PlanList />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});