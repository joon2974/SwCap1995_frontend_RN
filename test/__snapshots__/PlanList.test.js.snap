// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlanList /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
      "width": 750,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "width": 750,
      }
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": 750,
          },
          undefined,
        ]
      }
    />
    <RCTScrollView
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReached={[Function]}
      onEndReachedThreshold={1}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "marginTop": 10,
          "width": 750,
        }
      }
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;
